src/components/test/ProgramRecordsList.test.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import MockAdapter from 'axios-mock-adapter';
+import { Factory } from 'rosie';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getConfig } from '@edx/frontend-platform';
+import {
+  render, screen, cleanup, initializeMockApp,
+} from '../../setupTest';
+import ProgramRecordsList from '../ProgramRecordsList';
+import programListRecordsFactory from './__factories__/programListRecords.factory';
+
+describe('Program records list', () => {
+  beforeAll(async () => {
+    await initializeMockApp();
+  });
+  beforeEach(() => jest.resetModules);
+  afterEach(cleanup);
+
+  it('renders the component', () => {
+    render(<ProgramRecordsList />);
+    expect(screen.getByText('My Learner Records')).toBeTruthy();
+  });
+
+  it('it should display a link to the user\'s Profile', () => {
+    render(<ProgramRecordsList />);
+    expect(screen.getByText('Back to My Profile')).toBeTruthy();
+  });
+
+  it('it should have a help section', () => {
+    render(<ProgramRecordsList />);
+    expect(screen.getByText('Questions about Learner Records?')).toBeTruthy();
+  });
+});
+
+describe('Program list data', () => {
+  beforeAll(async () => {
+    await initializeMockApp();
+  });
+  afterEach(() => {
+    cleanup();
+    Factory.resetAll();
+  });
+
+  it('should display records when data is present', async () => {
+    const axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    axiosMock
+      .onGet(`${getConfig().CREDENTIALS_BASE_URL}/records/api/v1/program_records/`)
+      .reply(200, programListRecordsFactory.build());
+    render(<ProgramRecordsList />);
+    expect(await screen.findByText('edX | Completed')).toBeTruthy();
+    expect(await screen.findByText('edX | Partially Completed')).toBeTruthy();
```

Comment:
Do you think it's overkill if we add an additional check to make sure the `View Program Record` buttons exist as well?

This button shuffles learners over to the Program Record page and I think we'd want to make sure that piece of the program record component is being rendered.

Reply:
Good call! I'll get on that